### PR TITLE
strformat_lab.py

### PR DESCRIPTION
**Pull request by ikempcode33**
Final string format lab.

### CODE REVIEW

**Comment by ikempcode33** [comment]
Mailroom part 1 is ready for grading

File: students/Isabella_Kemp/lesson03/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    # For loop going through donors and calculating sum, number of donations, and average donations
+    # Then formatting them correctly into the table. Returns to main menu at the end.
+    for donor in sort_donor:
+        total = sum(donor[1])
+        num_donations = len(donor[1])
+        average = total/num_donations
+        print('{:<20} ${:>11,.2f}{:>13}            ${:>11,.2f}'.format(donor[0], total, num_donations, average))
+    return #return to main menu
+
+#define sort key
```
Comment (charlesrobison) [inline students/Isabella_Kemp/lesson03/mailroom.py]: Missing whitespace: # define...

File: students/Isabella_Kemp/lesson03/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+# creates a report of donor name, total donated, number of donations, and average
+# donation amount in each row
+
+def report():
+    print("\n{:<19}| {:<13} | {:<13} | {:>13}".format('Donor Name', 'Total Donated',
+                                                'Number of Donations',
+                                                'Avg. Donation Amount'))
+    print("-"*80)
+    # Takes donors and sorts it representing each item in the list
+    sort_donor = sorted(donors, key = sort_key,reverse = True)
```
Comment (charlesrobison) [inline students/Isabella_Kemp/lesson03/mailroom.py]: Spacing off on this section (linter will catch this):
sort_donor = sorted(donors, key=sort_key, reverse=True)

File: students/Isabella_Kemp/lesson03/mailroom.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            thank_you()
+        elif response == "2":
+            report()
+        elif response == "3":
+            exit_menu()
+        else:
+            print("Main menu input is invalid, enter 1 2 3 only please")
+
+if __name__ == "__main__":
+    main()
```
Comment (charlesrobison) [inline students/Isabella_Kemp/lesson03/mailroom.py]: Works great - nice work!

**Comment by ikempcode33** [comment]
Hi Charles!

 

I know you mentioned on my last submission to download a linter, and I did download pylint since I am using the pycharm IDE. However, do you know of any linters that I can download that would work? Mine does not seem to be detecting any of these spacing errors, etc. I like the pycharm IDE, and find it helpful, however I still have not gotten a linter to work on it. I have also activated all PEP8 setting within pycharm. 

 

Thank you,

 

Isabella Kemp 

 

 

 

----------------------------------------------------

Isabella Kemp 

Department of Bioengineering

Undergraduate Researcher

University of Washington Class of 2021

Ikemp33@uw.edu

**Comment by natasha-aleksandrova** [comment]
@ikempcode33 your pycharm already comes with a linter so you don't need any additional linter.

Pycharm should underline lines that have issues or show warn icon next to the line.

Are you not seeing those?

**Comment by ikempcode33** [comment]
I’m not seeing those for some reason, and I tried turning on all the settings for warnings in the preferences.

**Comment by ikempcode33** [comment]
Hi,

 

I am currently on macOS Catalina version 10.15.2. I have set up all the PEP8 standards inside of PyCharm to give me red errors but it doesn’t seem to work at all. I get no inspections on my code and I have said it back to defaults and it still doesn’t work. I am using PyCharm EDU 2019.3. 

 

Isabella 

 

----------------------------------------------------

Isabella Kemp 

Department of Bioengineering

Undergraduate Researcher

University of Washington Class of 2021

Ikemp33@uw.edu

**Pull request merged**